library/std/src/io/stdio.rs

```diff
@@ -1105,7 +1113,51 @@ pub fn _print(args: fmt::Arguments<'_>) {
 #[doc(hidden)]
 #[cfg(not(test))]
 pub fn _eprint(args: fmt::Arguments<'_>) {
-    print_to(args, stderr, "stderr");
+    struct StackBuf<'a, W> {
+        buf: BorrowedBuf<'a>,
+        stack_offset: usize,
+        inner: W,
+    }
+
+    impl<W: Write> Write for StackBuf<'_, W> {
+        fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+            if buf.len() >= self.buf.capacity() || buf.len() > self.buf.unfilled().capacity() {
+                self.inner
+                    .write_vectored(&[
+                        IoSlice::new(&self.buf.filled()[self.stack_offset..]),
+                        IoSlice::new(buf),
+                    ])
+                    .map(|written| {
+                        let offset_written = written + self.stack_offset;
+                        let stack_stored = self.buf.filled().len();
+                        if offset_written >= stack_stored {
+                            self.stack_offset = 0;
+                            self.buf.clear();
+                            offset_written - stack_stored // Num passthrough bytes written
+                        } else {
+                            self.stack_offset += written;
+                            0
```

Comment:
From the docs of [`Write::write`](https://doc.rust-lang.org/stable/std/io/trait.Write.html#tymethod.write):

> A return value of Ok(0) typically means that the underlying object is no longer able to accept bytes and will likely not be able to in the future as well, or that the buffer provided is empty.